### ISSUE TITLE
[Bug]: full page reload if options omited solidjs router integration

### ISSUE THREAD

**Issue by XantreDev** [Telegram-Mini-Apps/tma.js#302]
### Telegram Application

Telegram Desktop

### Describe the Bug

Confusing default behavior of `initNavigator` and `BrowserNavigator` if options is omitted

### To Reproduce

init navigator without options and navigate with `<A />` component
```ts
const navigator = initNavigator("app-navigator-state");
void navigator.attach();

const Router = createRouter(navigator);
```

### Expected Behavior

I think default behaviour must be like with this code
```ts
const navigator = initNavigator("app-navigator-state", {
  hashMode: "default",
});
```

**Comment by XantreDev** [comment]
I think 'default' is a good default xD.
null as value to turn it off is also fine. (maybe it can be a string like 'none')

Probably it's not even about this property, it's more about the navigation type, in case without options it works like MPA navigator (it's strange in SPAs) or like SPA navigator
I think here type unions can be used. It can that accept different variants of hash prop depending on navigation type
`{type:'spa',...data} | {type:'mpa',...data}`

**Comment by heyqbnk** [comment]
Not really sure about `spa` and `mpa` as long as a big part of developers don't really know what it is. Moreover, it is a bit more wide term to describe the navigator behavior.

Ok, we can leave it like that:

```ts
interface Options {
  /** @default 'default' **/
  hashMode?: null | 'default' | 'slice';
}
```

In case, you want to turn off hash navigation, just pass `hashMode: null`. It seems a bit strange for me (disabling something this way), but I am sure there will be almost no developers using it at all.

**Comment by heyqbnk** [comment]
Released in #317. Thanks for report!
